netket/optimizer/qgt/qgt_onthefly.py

```diff
@@ -178,7 +186,7 @@ def _to_dense(self: QGTOnTheFlyT) -> jnp.ndarray:
     """
     Npars = nkjax.tree_size(self._params)
     I = jax.numpy.eye(Npars)
-    out = jax.vmap(lambda x: self @ x, in_axes=0)(I)
+    _, out = jax.lax.scan(lambda _, x: (None, self @ x), None, I)
```

Comment:
The problem is that the scan in vmap_chunked does not have a jax batching rule (i.e. the thing telling jax how to push down the vmap). 
When we are using the batched stuff to reduce memory consumption we would want to do a scan like this anyway. I think with some effort it might be possible to write a jax batching rule that achieves that, keeping the vmap for the implementation without chunks.

It was my impression that if people want the matrix they can use dense or pytree, but maybe the usecase for this is for many samples and few param

Reply:
can we at least have a switch? `if self.chunking: scan else vmap`

Reply:
actually it's the linear call which does not have the rule, but I'll just add your workaround for the time being